projects/igniteui-angular/src/lib/tree/tree.component.ts

```diff
@@ -67,47 +85,86 @@ export class IgxTreeComponent implements IgxTree, OnInit, AfterViewInit, OnDestr
     @Output()
     public nodeCollapsed = new EventEmitter<ITreeNodeToggledEventArgs>();
 
+    // TODO: should we remove this thus checkbox aren't templatable
     @ContentChild(IgxTreeSelectMarkerDirective, { read: TemplateRef })
     public selectMarker: TemplateRef<any>;
 
     @ContentChild(IgxTreeExpandIndicatorDirective, { read: TemplateRef })
     public expandIndicator: TemplateRef<any>;
 
     @ContentChildren(IgxTreeNodeComponent, { descendants: true })
-    private nodes: QueryList<IgxTreeNodeComponent<any>>;
+    public nodes: QueryList<IgxTreeNodeComponent<any>>;
 
     public id = `tree-${init_id++}`;
 
-    constructor(private selectionService: IgxSelectionAPIService, private treeService: IgxTreeService) {
+    private _selection: IGX_TREE_SELECTION_TYPE = IGX_TREE_SELECTION_TYPE.None;
+
+    constructor(private selectionService: IgxTreeSelectionService, private treeService: IgxTreeService) {
+        this.selectionService.register(this);
         this.treeService.register(this);
     }
 
-    public expandAll(nodes: IgxTreeNode<any>[]) {}
-    public collapseAll(nodes: IgxTreeNode<any>[]) {}
-    public selectAll(nodes: IgxTreeNode<any>[]) {}
+    public expandAll(nodes: IgxTreeNode<any>[]) { }
+    public collapseAll(nodes: IgxTreeNode<any>[]) { }
+
+    /**
+     * Select all nodes if the nodes collection is empty. Otherwise, select the nodes in the nodes collection.
+     *
+     * @example
+     * ```typescript
+     *  const arr = [
+     *      this.tree.nodes.toArray()[0],
+     *      this.tree.nodes.toArray()[1]
+     *  ];
+     *  this.tree.selectAll(arr, true);
+     * ```
+     * @param nodes: IgxTreeNodeComponent<any>[]
+     * @param clearPrevSelection: boolean; if true clears the current selection
+     */
+    public selectAll(nodes?: IgxTreeNodeComponent<any>[], clearPrevSelection?: boolean) {
```

Comment:
Bit of a nit pick, but I'd prefer `clearPrevSelection` to be explicitly `false` instead of `undefined`
```suggestion
    public selectAll(nodes?: IgxTreeNodeComponent<any>[], clearPrevSelection = false) {
```